nmdc_automation/re_iding/base.py

```diff
@@ -51,6 +51,19 @@ def __init__(self, api_client: NmdcRuntimeApi, data_dir: str,
             template_file = NAPA_TEMPLATE
         with open(template_file, "r") as f:
             self.workflow_template = yaml.safe_load(f)["Workflows"]
+        # collector to track ID changes as (type, old_id, new_id)
+        self.id_changes = []
+
+    def write_id_changes(self, study_id: str):
+        """
+        Write ID changes to a file.
+        """
+        filename = f"{study_id}_id_changes.tsv"
+        with open(filename, "w") as f:
+            f.write("type\told_id\tnew_id\n")
```

Comment:
I think you can use Python's built-in `csv` module to write TSV (by setting the delimiter to `'\t'`) and that that will take care of escaping quoted data that happens to contain the delimiter character (e.g. `foo\t"bar1\tbar2"\tbaz`).

https://docs.python.org/3/library/csv.html